### DISCUSSION TITLE
Remove `size` from geostyler-openlayers-parser

### DISCUSSION THREAD

**Discussion by KaiVolland** [geostyler/geostyler#1752] [Polls]
OpenLayers does not offer a way to set the size of an [ol/style/Icon](https://openlayers.org/en/latest/apidoc/module-ol_style_Icon.html).

When using the `size` property, a piece of the existing image can be cut out an displayed but it won't shrink or grow the image to the expected result.

IMHO we should remove the `size` property from the `geostyler-openlayers-parser` and add it to unsupportedProperties instead of applying it to `scale`.

Any thoughts @geostyler/devs ?

**Comment by KaiVolland** [comment]
It would theoretically be possible to get the image size with a package like https://github.com/image-size/image-size and calculate the `scale` out of it. But that would be asynchronous.

**Reply by marcjansen** [reply]
Can we try how far we can get with that?

**Comment by jansule** [comment]
> IMHO we should remove the size property from the geostyler-openlayers-parser and add it to unsupportedProperties instead of applying it to scale.

Would that mean that we do not have any way of adjusting the size using the openlayers parser? Upscaling should generally work as expected, right?

**Reply by KaiVolland** [reply]
Correct there would be no possibility to set the size of an Icon in the openlayers-parser. I don't know what you mean with upscaling.

**Reply by jansule** [reply]
Increasing the size

**Comment by slafayIGN** [comment]
Hi. I think it's better not to lose any styling possibility in openlayers-parser even if it includes "tricks" like switching from size to scale or passing an opacity as an alpha channel in a rgba color.

Maybe it lacks a document where the styling capabilities of every style parser are compared ?

Also maybe `size` could be described more accurately here : https://geostyler.github.io/geostyler-style/docs/master/interfaces/IconSymbolizer.html#size

For instance, the SLD doc states : 
> `<size>` Specifies the size of the symbol, in pixels. When used with an image file, this specifies the height of the image, with the width being scaled accordingly. if omitted the native symbol size is used.

There may be also an issue with the mapbox-parser where `size` is mapped with `icon-size`... which is actually more a like a `scale` according to Mapbox doc : https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-icon-size
> Scales the original size of the icon by the provided factor. The new pixel size of the image will be the original pixel size multiplied by icon-size. 1 is the original size; 3 triples the size of the image.

**Comment by KaiVolland** [comment]
If mapbox and ol have a 'scale' we may should add it to geostyler-style too:
- Less confusion
- more precise
🤔

**Reply by slafayIGN** [reply]
Yes less confusion and more precise...

But on the other hand, this easier way of making things would make style format conversions less relevant. We would lose rendering capabilities on conversion.

**Reply by KaiVolland** [reply]
Wouldn't we add more capabilities of rendering? Currently we're bound to pixels, if we add `scale` we would support more ways to render an `Icon`.
Or did i get you wrong?
